### PR TITLE
[wip] allow updater to recover from master breaking changes

### PR DESCRIPTION
**Pull request by ktsaou**
##### Summary

This PR will allow the `netdata-updater.sh` script to:

1. fetch the repo tags
2. update netdata, when there is a breaking change to the repo

##### Additional Information

```paste below

```

### CODE REVIEW

File: netdata-installer.sh [hunk 1]
```diff
@@ -1197,7 +1197,23 @@ update() {
...
+        # make sure we have the latest tags of the repo (including tags)
+        git fetch --tags  >&3 2>&3 || fetched=0
+
+        # pull any updates
+        git pull >&3 2>&3 || pulled=0
+
+        if [ \$fetched -eq 1 ] && [ \$pulled -eq 0 ]
+        then
+            # we fetched the repo, but we failed to pull changes
+            # so, reset the local clone to match the origin
```
[the remark this answers is not in the document]
Comment (ktsaou) [reply netdata-installer.sh]: added it

File: netdata-installer.sh [hunk 2]
```diff
@@ -1197,7 +1197,30 @@ update() {
...
+
+        # pull any updates
+        git pull >&3 2>&3 || pulled=0
+
+        if [ \$pulled -eq 0 ]
+        then
+            error "FAILED TO PULL REPO FROM GITHUB... trying full re-sync..."
+
+            info "fetching latest repo data from github..."
+            git fetch --all >&3 2>&3 || failed "FAILED TO FETCH THE LATEST SOURCE FROM GITHUB"
```
Comment (paulfantom) [inline netdata-installer.sh]: `fetch --all` won't fetch tags. `fetch --tags` needs to be run.

**Comment by paulfantom** [comment]
This is unnecessary overcomplicated. Simple sequence of:
```
git stash
git fetch --all
git fetch --tags
git checkout origin/master
git reset --hard origin/master
```

solves all cases without using bash variables and `git pull`. As a bonus it also preserves local changes in a stash (can be reapplied by `git stash pop`)

Remember that `git pull` is basically a wrapper on `git fetch && git merge FETCH_HEAD`.

**Comment by paulfantom** [comment]
Implemented in #4558

**Pull request closed without merging**